lsp/source/lib/typescript-service.mts

```diff
@@ -416,17 +420,32 @@ function TSService(projectURL = "./") {
   const projectRequire = createRequire(projectURL)
 
   // Use Civet from the project if present
-  let Civet: typeof BundledCivetModule
+  let Civet = BundledCivetModule
+  let CivetConfig = BundledCivetConfigModule
   try {
     const civetPath = "@danielx/civet"
     Civet = projectRequire(civetPath)
-
+    CivetConfig = projectRequire(`${civetPath}/config`)
     console.info(`LOADED PROJECT CIVET: ${path.join(projectURL, civetPath)} \n\n`)
   } catch (e) {
     console.info("USING BUNDLED CIVET")
-    Civet = BundledCivetModule
   }
 
+  let civetConfig: CompileOptions = {}
+  CivetConfig.findConfig(projectPath).then(async configPath => {
+    if (configPath) {
+      console.info("Loading Civet config @", configPath)
+      const envLsp = process.env.CIVET_LSP
+      process.env.CIVET_LSP = "true"
```

Comment:
Setting `process.env.CIVET_LSP` isn't safe in concurrent situations. It would be better to pass an option to `loadConfig`.

Reply:
How do you suggest passing information to the config file? `loadConfig` uses an `await import`. Would a global `const CIVET_LSP: boolean` or `const CIVET_CONFIG_INFO: { lsp?: boolean, ... }` suffice?

I'm also in favor of just removing the CIVET_LSP environment variable entirely, to always have CLI and LSP behave the same.

Reply:
You could pass a second optional parameter to `loadConfig` that way each call to load will have it's own copy even if they are concurrent. Removing `CIVET_LSP` is also fine for now.

Thanks for your work on this PR!

Reply:
Removed it